go-controller/pkg/ovn/pods.go

```diff
@@ -183,6 +185,14 @@ func getRoutesGatewayIP(pod *kapi.Pod, gatewayIPnet *net.IPNet) ([]util.PodRoute
 			route.NextHop = gatewayIPnet.IP
 			routes = append(routes, route)
 		}
+		if namespaceExternalGW != "" {
+			if net.ParseIP(namespaceExternalGW) == nil {
```

Comment:
This seems kind of out of place here... and it's awkward that we're basically saying that every place that needs to look at the annotation needs to first validate it.

I think it would be better to, instead of having `nsInfo.annotations`, have either `nsInfo.externalGatewayIP` or `nsInfo.hasExternalGateway`, which would only be set if the annotation was both present and valid.

Reply:
Yeah, that's a better solution, though right now there's just one place on the master side that uses it, so not sure how useful that would be. But I like it better.